### DISCUSSION TITLE
Error of recompiling my application: 'not a registered file'

### DISCUSSION THREAD

**Discussion by lixinjie-1996** [idaholab/moose#25793] [Q&A General]
Hi!

I'm trying to add my own .C and .h file into my application called 'newapp', but it doesn't work. The error is :
```
*** ERROR ***
A 'StressSpectral' is not a registered object.

'StressSpectral.C' file and 'StressSpectral.h' file are created by myself.

Here is what I do:
- copy the .C file to newapp/src/kernels
- copy the .h file to newapp/include/kernels
- do 'make clean' to the app
- then recomplie my newapp with 'make -j 4'
```
I can see that the app recompiled with 'StressSpectral.C' file and 'StressSpectral.h' file.
After the recompiling was finished, I tried to run an example from moose (crack2d.i in combined modulus) and it worked well.
But when I try to run another input file with StressSpectral.C and StressSpectral.h in the new app, I got this error:
A 'StressSpectral' is not a registered object.

Do you have any idea about how to solve this problem?

Thank you.

**Comment by GiudGiud** [comment]
Hello

Can you paste the `registerMooseObject` line in your kernel here?
And what is the name of your app in the src/base folder?

**Reply by lixinjie-1996** [reply]
Hi!

Here is it:
registerMooseObject("TensorMechanicsApp", StressSpectralViscoPlastic);

So 'StressSpectralViscoPlastic' is the name of my .C and .h file

and the name of my app is 'bokeloh'
Here is the paths of the .C and .h file:
~/projects/bokeloh/src/kernels/StressSpectralViscoPlastic.C
~/projects/bokeloh/include/kernels/StressSpectralViscoPlastic.h

**Reply by GiudGiud** [reply]
The names need to match. If you register it to stressspectralviscoplastic you need to call it that name in the input file

**Reply by GiudGiud** [reply]
For the name of the app you should register it to your app not the tensor mechanics app. 

If your app includes tensor mechanics then it should still work

**Reply by lixinjie-1996** [reply]
I've changed it to:
registerMooseObject("bokelohApp", StressSpectralViscoPlastic);

and I've verified that all the modulus in the Makefile of my app bokeloh has been turned on with 'yes'.
also, tried to do 'make clobberall ' and then 'make -j 4' several times.

However, the same problem still persists.
A 'StressSpectralViscoPlastic' is not a registered object.

**Reply by GiudGiud** [reply]
in the src/base/bokelohApp.C file what is the spelling of the app name exactly

**Reply by lixinjie-1996** [reply]
Sorry for my late reply.
I changed my computer system from Centos to Ubuntu and now this problem does not appear anymore. It's a bit mysterious but the error is fixed anyway.
Thank you so much for your help.
